banyand/liaison/grpc/property.go

```diff
@@ -762,3 +772,115 @@ type repairInProcessKey struct {
 	entity  string
 	modTime int64
 }
+
+// sortProperties sorts properties based on the specified order_by field.
+func sortProperties(properties []*propertyv1.Property, orderBy *propertyv1.QueryOrder) []*propertyv1.Property {
+	if len(properties) == 0 || orderBy == nil || orderBy.TagName == "" {
+		return properties
+	}
+
+	isDesc := orderBy.Sort == modelv1.Sort_SORT_DESC
+
+	sort.Slice(properties, func(i, j int) bool {
+		// Extract tag values
+		valI := getPropertyTagValue(properties[i], orderBy.TagName)
+		valJ := getPropertyTagValue(properties[j], orderBy.TagName)
+
+		// Handle nil values - put them at the end
+		if valI == nil {
+			return false
+		}
+		if valJ == nil {
+			return true
+		}
+
+		// Compare values
+		cmp := comparePropertyTagValues(valI, valJ)
+
+		if isDesc {
+			return cmp > 0
+		}
+		return cmp < 0
+	})
+
+	return properties
+}
+
+// getPropertyTagValue retrieves a tag value by name from a property.
+func getPropertyTagValue(prop *propertyv1.Property, tagName string) *modelv1.TagValue {
```

Comment:
Add the sorting tag as an independent tag to propertyv1.Property. The sorting tag might not be in the projection. Using such a tag helps us handle sorting without returning the tag through the projection tag list.